docs/docs/03-light-clients/04-wasm/03-integration.md

```diff
@@ -0,0 +1,240 @@
+- Add the option created in the previous step to a slice and [pass it to the `x/wasm NewKeeper` constructor function](https://github.com/CosmWasm/wasmd/blob/db93d7b6c7bb6f4a340d74b96a02cec885729b59/x/wasm/keeper/keeper_cgo.go#L36).
+- Pass the pointer to the Wasm VM instance to `08-wasm` [NewKeeperWithVM constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L33-L38).
+
+The code to set this up would look something like this:
+
+```go
+// app.go
+import (
+  ...
+  wasmvm "github.com/CosmWasm/wasmvm"
+  wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+  ...
+)
+
+...
+
+// instantiate the Wasm VM with the chosen parameters
+wasmer, err := wasmvm.NewVM(
+  dataDir, 
+  availableCapabilities, 
+  contractMemoryLimit,
+  contractDebugMode, 
+  memoryCacheSize,
+)
+if err != nil {
+  panic(err)
+}
+
+// create an Option slice (or append to an existing one)
+// with the option to use a custom Wasm VM instance
+wasmOpts = []wasmkeeper.Option{
+  wasmkeeper.WithWasmEngine(wasmer),
+}
+
+// the keeper will use the provided Wasm VM instance,
+// instead of instantiating a new one
+app.WasmKeeper = wasmkeeper.NewKeeper(
+  appCodec,
+  keys[wasmtypes.StoreKey],
+  app.AccountKeeper,
+  app.BankKeeper,
+  app.StakingKeeper,
+  distrkeeper.NewQuerier(app.DistrKeeper),
+  app.IBCFeeKeeper, // ISC4 Wrapper: fee IBC middleware
+  app.IBCKeeper.ChannelKeeper,
+  &app.IBCKeeper.PortKeeper,
+  scopedWasmKeeper,
+  app.TransferKeeper,
+  app.MsgServiceRouter(),
+  app.GRPCQueryRouter(),
+  wasmDir,
+  wasmConfig,
+  availableCapabilities,
+  authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+  wasmOpts...,
+)
+
+app.WasmClientKeeper = wasmkeeper.NewKeeperWithVM(
+  appCodec,
+  keys[wasmtypes.StoreKey], 
+  authtypes.NewModuleAddress(govtypes.ModuleName).String(),
+  wasmer, // pass the Wasm VM instance to `08-wasm` keeper constructor
+)
+...
+```
+
+### If `x/wasm` is not present
+
+If the chain does not use [`x/wasm`](https://github.com/CosmWasm/wasmd/tree/main/x/wasm), even though it is still possible to use the method above from the previous section
+(e.g. instantiating a Wasm VM in app.go an pass it to 08-wasm's [`NewKeeperWithVM` constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L33-L38), since there would be bi need in this case to share the Wasm VM instance with another module, you can use the [`NewKeeperWithConfig`` constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L52-L57) and provide the Wasm VM configuration parameters of your choice instead. A Wasm VM instance will be created in`NewKeeperWithConfig`. The parameters that can set are:
```

Comment:
```suggestion
(e.g. instantiating a Wasm VM in app.go an pass it to 08-wasm's [`NewKeeperWithVM` constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L33-L38), since there would be the need in this case to share the Wasm VM instance with another module, you can use the [`NewKeeperWithConfig` constructor function](https://github.com/cosmos/ibc-go/blob/c95c22f45cb217d27aca2665af9ac60b0d2f3a0c/modules/light-clients/08-wasm/keeper/keeper.go#L52-L57) and provide the Wasm VM configuration parameters of your choice instead. A Wasm VM instance will be created in`NewKeeperWithConfig`. The parameters that can set are:
```

Reply:
I think the `bi` is supposed to be a `no` rather than `the`